core/src/test/java/hudson/slaves/SlaveComputerTest.java

```diff
@@ -0,0 +1,65 @@
+package hudson.slaves;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Stephen Connolly
+ */
+public class SlaveComputerTest {
+    @Test
+    public void isRelativePathUnix() {
+        assertThat("/", not(aRelativePath()));
+        assertThat("/foo/bar", not(aRelativePath()));
+        assertThat("/foo/../bar", not(aRelativePath()));
+        assertThat("", aRelativePath());
+        assertThat(".", aRelativePath());
+        assertThat("..", aRelativePath());
+        assertThat("./foo", aRelativePath());
+        assertThat("./foo/bar", aRelativePath());
+        assertThat("./foo/bar/", aRelativePath());
+    }
+
+    @Test
+    public void isRelativePathWindows() {
+        assertThat("\\", aRelativePath());
+        assertThat("\\foo\\bar", aRelativePath());
+        assertThat("\\foo\\..\\bar", aRelativePath());
+        assertThat("", aRelativePath());
+        assertThat(".", aRelativePath());
+        assertThat(".\\foo", aRelativePath());
+        assertThat(".\\foo\\bar", aRelativePath());
+        assertThat(".\\foo\\bar\\", aRelativePath());
+        assertThat("\\\\foo", aRelativePath());
+        assertThat("\\\\foo\\", not(aRelativePath()));
```

Comment:
I know but I wanted a faster test and this is better than anything else we had... it can be improved if somebody feels the need ;-)